resolver/src/main/java/io/netty/resolver/RoundRobinInetAddressResolver.java

```diff
@@ -64,7 +64,7 @@ public void operationComplete(Future<List<InetAddress>> future) throws Exception
                     if (numAddresses > 0) {
                         // if there are multiple addresses: we shall pick one by one
                         // to support the round robin distribution
-                        promise.setSuccess(inetAddresses.get(index.getAndIncrement() % numAddresses));
+                        promise.setSuccess(inetAddresses.get(toIndex(counter.getAndIncrement(), numAddresses)));
```

Comment:
@normanmaurer This method has one problem. It will change a direction of selecting servers after changing sign. It's not awfully bad, but it's changing behavior from one stable behavior to another one in an unpredictable moment of time (overflow, crossing 0). If this cause any problems, which is not very probable, but still probably - would be hard to find and debug. So, should be good to avoid, even before current code is not harder to read and maintain.
One more - this code is just the same what's doing in Collections.rotate() with parameter in the JDK code.